rfcs/0000-since-parameter.md

```diff
@@ -0,0 +1,141 @@
+
+Today, BFD only supports returning all EOB resources associated with a single beneficiary. EOB calls return more than 5 years of beneficiary data, where only the last weeks of data is needed. This behavior is highly inefficient for the bulk export calls that happen weekly. On average, each call is returning 260 times as much information as is needed. 
+
+Early feedback from both BCDA and DPC customers have nearly unanimously pointed out the need for _since parameter support \[[2](#ref2)\]. For BCDA, where an export operation can take many hours and result in 100's GB of data, BCDA customers have stated that they need '_since' support to move to production. BB 2.0 app developers would like a similar feature as well. 
+
+## Proposed Solution
+
+This proposal adds 3 changes to the BFD API that are needed for downstream partners to implement the `_since` parameter. 
+
+1. The `lastUpdated` metadata field of EOB, Patient, and Coverage FHIR resources contains the time they were written to the master DB by the ETL process. 
+2. The search operation of the EOB, Patient, and Coverage resources support `_lastUpdated` query parameter. When specified, the search operation filters resources against the passed in timestamp. 
+3. The API adds a feed containing the start and finish times of the ETL job into BFD and a list of beneficiaries updated in the ETL. This feed enables BFD partners to correctly and efficiently implement the `_since` parameter. 
+
+Using the new information provided by the BFD, a partner can implement `_since` exports. To improve the efficiency of the export operation and its performance, partners must avoid calling the BFD if a particular beneficiary doesn't have an updated resource (see [Since Implementors Details](#since-implementors-details)). The following sequence diagram shows how this interaction should work. 
+
+![Bulk ](https://www.websequencediagrams.com/files/render?link=zfMUJyQaf18DNUb6IQoPN2EBeq9tMctYXXupx6T5Co8gB3t9ysmhat0ToalxZ6p2)
+
+### BFD API Details
+
+All the proposed API changes follow the FHIR specification. The first improvement is to add the `lastUpdated` to the metadata field of a resource. The current implementation returns the time the that the request was made. The proposal alters this field to be the timestamp that the ETL process wrote to the master DB. Like all instant timestamps, this timestamp must include timezone \[[4](#ref4)\]. 
+
+The second change is to support the `_lastUpdated` query parameter for resource searches \[[5](#ref5)\]. FHIR specifies a set of comparison operators to go along with this filter. Although bulk exporters will only use the `gt` or greater-than operator, a complete set will be useful for other use-cases. 
```

Comment:
Is there behavior that can be described here that would allow the BFD to protect itself from misbehaving consumers? Can it be made cheap for the BFD to return nothing if no resources fall within the `_lastUpdated` query?

Reply:
Good point. The BFD server could keep the list of updated beneficiaries itself. So the database scan can be avoided for beneficiaries without a recent update. I like this idea a lot.